src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java

```diff
@@ -149,8 +149,29 @@ private void onPartitionsRevoked(Collection<TopicPartition> partitions) {
         log.debug("onPartitionsRevoked {}", partitions);
         if (!partitions.isEmpty()) {
             // It is safe to use the consumer here since we are in a poll()
-            if (ackMode != AckMode.ATMOST_ONCE)
+            if (ackMode != AckMode.ATMOST_ONCE) {
                 commitEvent.runIfRequired(true);
+                long maxDelayRebalance = receiverOptions.maxDelayRebalance().toMillis();
+                if (isActive.get() && maxDelayRebalance > 0) {
+                    long interval = receiverOptions.commitIntervalDuringDelay();
+                    int inPipeline = commitEvent.commitBatch.getInPipeline();
+                    if (inPipeline > 0 || this.awaitingTransaction.get()) {
+                        long end = maxDelayRebalance + System.currentTimeMillis();
+                        do {
+                            try {
+                                log.debug("Rebalancing; waiting for {} records in pipeline", inPipeline);
+                                Thread.sleep(interval);
```

Comment:
would it be better to schedule so task `interval` milliseconds in the future, rather than sleep/blocking ? or this is guaranteed to be executed on a `Thread` that can be blocked?

Reply:
From what I know, rebalance happens on the consumption thread, and during rebalance, consumer is not polling messages. I think its better to block the thread while the thread pool continues to process and acknowledge messages. This way we prevent redundant thread pool tasks transfers*

Reply:
We must block it to prevent Kafka from completing the rebalance while the pipeline is drained.

I believe it is a blockable thread because `commitSync()` (which is optionally called) blocks.